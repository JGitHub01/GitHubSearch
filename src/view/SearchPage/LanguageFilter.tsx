const languages = [
  'JavaScript',
  'Java',
  'Python',
  'TypeScript',
  'C#',
  'C',
  'PhP',
  'Shell',
  'Ruby'].sort();
interface ILanguageFilterProps {
  language?: string
  onLanguageChange?: (language: string) => void
}
export default function LanguageFilter(props: ILanguageFilterProps) {
  const { language, onLanguageChange } = props;
  return (
    <div className='github-search__language' data-testid='search-language'>
      <label htmlFor='language'>Language:</label>
      <select name='language' data-testid='language-list'
        value={language}
        onChange={(e) => {
          onLanguageChange && onLanguageChange(e.target.value);
        }}
      >
        {languages.map((lan, idx) => {
          return (<option key={idx} value={lan.toLowerCase()}>{lan}</option>);
        })}
      </select>
    </div>
  );
}
