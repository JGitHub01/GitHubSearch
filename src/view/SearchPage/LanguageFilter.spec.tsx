import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import LanguageFilter from './LanguageFilter';

describe('LanguageFilter', () => {
  it('Should render the default snapshot view', () => {
    const { getByTestId } = render(<LanguageFilter />);
    const lanFilter = getByTestId('search-language');
    expect(lanFilter).toMatchSnapshot();
  });

  it('Should render the view with a set language', () => {
    const { getByTestId } = render(<LanguageFilter language="python" />);
    const lanList = getByTestId('language-list');
    expect(lanList).toHaveValue("python");
  });

  it('Should call onLanguageChange with the new language when a new language is selected', () => {
    const callback = jest.fn();
    const { getByText, getByTestId } = render(<LanguageFilter onLanguageChange={callback} />);
    const lanList = getByTestId('language-list');
    const phpOption = getByText('PhP');
    userEvent.selectOptions(lanList, phpOption);
    expect(callback).toBeCalledWith('php');
  });
});
