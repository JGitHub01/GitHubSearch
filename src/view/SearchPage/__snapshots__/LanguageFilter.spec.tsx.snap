// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageFilter Should render the default snapshot view 1`] = `
<div
  class="github-search__language"
  data-testid="search-language"
>
  <label
    for="language"
  >
    Language:
  </label>
  <select
    data-testid="language-list"
    name="language"
  >
    <option
      value="c"
    >
      C
    </option>
    <option
      value="c#"
    >
      C#
    </option>
    <option
      value="java"
    >
      Java
    </option>
    <option
      value="javascript"
    >
      JavaScript
    </option>
    <option
      value="php"
    >
      PhP
    </option>
    <option
      value="python"
    >
      Python
    </option>
    <option
      value="ruby"
    >
      Ruby
    </option>
    <option
      value="shell"
    >
      Shell
    </option>
    <option
      value="typescript"
    >
      TypeScript
    </option>
  </select>
</div>
`;
